presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestQueries.java

```diff
@@ -9215,4 +9215,50 @@ public void testDefaultDecimalLiteralSwitch()
         assertEquals(doubleColumnResult.getTypes().get(0), DOUBLE);
         assertEquals(doubleColumnResult.getMaterializedRows().get(0).getField(0), 1.0);
     }
+
+    @Test
+    public void testInnerJoinEmptyRightTable()
+    {
+        MaterializedResult actual = computeActual("" +
+                "WITH small_part AS (SELECT * FROM (SELECT * FROM part LIMIT 20) WHERE RAND() < 0) " +
+                "SELECT lineitem.orderkey FROM lineitem INNER JOIN small_part ON lineitem.partkey = small_part.partkey");
+
+        assertEquals(actual.getRowCount(), 0);
+    }
+
+    @Test
+    public void testRightJoinEmptyRightTable()
+    {
+        MaterializedResult actual = computeActual("" +
+                "WITH small_part AS (SELECT * FROM (SELECT * FROM part LIMIT 20) WHERE RAND() < 0) " +
+                "SELECT lineitem.orderkey FROM lineitem RIGHT JOIN small_part ON lineitem.partkey = small_part.partkey");
+
+        assertEquals(actual.getRowCount(), 0);
+    }
+
+    @Test
+    public void testLeftJoinEmptyRightTable()
+    {
+        MaterializedResult lineItemRowCount = computeActual("" +
```

Comment:
Please inline this count, it is constant. Running this query do not test your feature.

Reply:
Updated.